src/sync/discovery.ts

```diff
@@ -1,14 +1,76 @@
-import { Subscription } from "../models";
 import { getRepositorySummary } from "./jobs";
 import enhanceOctokit from "../config/enhance-octokit";
 import { Application } from "probot";
-import { Repositories, SyncStatus } from "../models/subscription";
-import {LoggerWithTarget} from "probot/lib/wrap-logger";
+import Subscription, { Repositories, Repository, SyncStatus } from "../models/subscription";
+import { LoggerWithTarget } from "probot/lib/wrap-logger";
 import { sqsQueues } from "../sqs/queues";
+import GitHubClient from "../github/client/github-client";
+import { getCloudInstallationId } from "../github/client/installation-id";
 
 export const DISCOVERY_LOGGER_NAME = "sync.discovery";
 
-export const discovery = (app: Application) => async (job, logger: LoggerWithTarget) => {
+const mapRepositories = (repositories: Repository[]): Repositories => {
+	return repositories.reduce((obj, repo) => {
+		obj[repo.id] = { repository: getRepositorySummary(repo) };
+		return obj;
+	}, {});
+};
+
+const resetNumberOfSyncedRepos = async (subscription: Subscription) => {
+	await subscription.updateSyncState({ numberOfSyncedRepos: 0 });
+};
+
+const updateSyncState = async (subscription: Subscription, repositories: Repository[]) => {
+	const repos = mapRepositories(repositories);
+	await subscription.updateSyncState({
+		repos
+	});
+};
+
+const checkHeaderForNextPage = (headers): boolean => {
```

Comment:
Github returns a link header attribute if there is another page (https://docs.github.com/en/rest/guides/traversing-with-pagination) there is exceptions where no link header will be sent too, but those exceptions don't have more data.
header looks like
header: {
...
links: ["<https://api.github.com/installation/repositories?per_page=1&page=2>; rel=\"prev\"", "<https://api.github.com/installation/repositories?per_page=1&page=2>; rel=\"next\""]
...
}

Reply:
headers can be typed with `AxiosResponseHeaders`